pavelib/prereqs.py

```diff
@@ -121,23 +124,49 @@ def prereq_cache(cache_name, paths, install_func):
         print '{cache} unchanged, skipping...'.format(cache=cache_name)
 
 
+@timeout(limit=60)
 def node_prereqs_installation():
     """
     Configures npm and installs Node prerequisites
     """
+    # NPM installs hang sporadically. Log the installation process so that we
+    # determine if any packages are chronic offenders.
+    shard_str = os.getenv('SHARD', None)
+    if shard_str:
+        npm_log_file_path = '{}/npm-install.{}.log'.format(Env.GEN_LOG_DIR, shard_str)
+    else:
+        npm_log_file_path = '{}/npm-install.log'.format(Env.GEN_LOG_DIR)
+    npm_log_file = open(npm_log_file_path, 'w')
+    npm_command = 'npm install --verbose'.split()
+
     cb_error_text = "Subprocess return code: 1"
 
     # Error handling around a race condition that produces "cb() never called" error. This
     # evinces itself as `cb_error_text` and it ought to disappear when we upgrade
     # npm to 3 or higher. TODO: clean this up when we do that.
```

Comment:
We're on npm 5.5.1...please ticket cleaning this up, including the test for this error handling

Reply:
https://openedx.atlassian.net/browse/TE-2716